lib/runtime/storage/trie.go

```diff
@@ -146,8 +146,13 @@ func (t *TrieState) MustRoot() common.Hash {
 	return hash
 }
 
-// Root returns the trie's root hash
+// Root is executed in the block finalisation
+// when it is wrapping everything and needs to ensure
+// the root hash matches the expected one, in this case
+// we commit the changeset we started in the beginning
 func (t *TrieState) Root() (common.Hash, error) {
+	t.CommitTransaction()
```

Comment:
I have some concerns here.
What if the runtime start a transaction `ext_storage_start_transaction` then commit that transaction `ext_storage_commit_transaction` and then ask for the root `ext_storage_root` ?
If I'm not wrong this `t.CommitTransaction()` will panic because of this [condition](https://github.com/ChainSafe/gossamer/blob/76741e489e4b7426e6b4e56699e7d6c32e536f6d/lib/runtime/storage/trie.go#L85-L87)

Reply:
The `ext_storage_root ` is only called in the wrap up of the block, and it happens only once in the whole block execution lifetime, added a warn to this function comment also removed calls to this function from other places in the code.

Reply:
Thanks to check that! ❤️